### PR TITLE
build: add community credits and about sections to changelog generation

### PR DESCRIPTION
**Pull request by sgtcoolguy**
**Description:**
We have code to generate a changelog from our conventional commits which works fairly well. However, it's missing some common items from our release notes, namely: community credits, an "about" section which details end of life/support for releases, and module info.

This PR aims to add the first two mentioned to the generated changelog process. When gathering commits this now also includes author email and name and filters those against "known" email addresses/domain used by employees. We then gather commits per unique "authorName" and add a credits section listing individuals with the list of commits/issues they helped out on. Please note that if the user has differing authorNames on different commits it's not smart enough to auto-merge them (i.e. we have some community members who must commit from separate machines who I know are the "same" person despite the commits being labelled differently).

Additionally, we have some fairly boilerplate support language at the top of our notes which list out the versions/dates of support changes. This codes in the determinations of the branch/versions involved and the dates based on the "release date" (date the changelog was generated).

As always, this is not the be-all, end-all of the release notes - but it should get us about 95% of the way and require some fairly minimal "massaging". Ideally it would also combine/cross-reference with the JIRA release notes generated to help improve the ticket title in preference to the commit subject (which may be lower-level/implementation specific).

### CODE REVIEW

File: changelog/config.js [hunk 1]
```diff
@@ -178,13 +228,54 @@ module.exports = {
 				};
 			});
 
+			// convert communityContributions from map to array of objects!
+			const blah = [];
```
Comment (ewanharris) [inline changelog/config.js]: Maybe just push onto `context.communityContributions` directly?

**Review by ewanharris — COMMENTED** [review]
Few comments, but this looks awesome!

I tried to run it locally but I think we're missing the `conventional-changelog-cli` package in the devDeps? `npm i -D conventional-changelog-cli` got it running for me

**Comment by ewanharris** [comment]
Here's the output of this is anyone is interested https://gist.github.com/ewanharris/a98ef148e2355b6d9e0b434994f5348c

File: changelog/config.js (hunk 1 above)
Comment (sgtcoolguy) [reply changelog/config.js]: done

File: changelog/templates/about.hbs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+## About this release
+
+{{#if isPatch~}}
+Titanium SDK {{version}} is a patch release of the SDK, addressing high-priority issues from previous releases.
+
+As of this GA release, the previous Titanium SDK patch release ({{eosBranch}}) is no longer supported. End of support for this version will be {{eosDate}} or until the next patch release. Note: major and minor releases continue to be supported according to their nominal lifetime.
+{{~else~}}
+  {{~#if isMajor~}}
```
[the remark this answers is not in the document]
Comment (sgtcoolguy) [reply changelog/templates/about.hbs]: yup, thanks

File: changelog/templates/about.hbs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+As of this release, Titanium SDK {{eosBranch}} will not be supported one calendar year ({{eosDate}}) from {{version}}'s release date.
+  {{~else~}}
+Titanium SDK {{version}} is a minor release of the SDK, addressing high-priority issues from previous releases.
+
+As of this release, Titanium SDK {{eosBranch}} will not receive updates more than six months after the release of {{version}} ({{eosDate}}). Any needed fixes will be in {{patchBranch}} or later supported releases within the {{majorBranch}} branch.
+  {{~/if~}}
+{{~/if}}
+
+See [Axway Appcelerator Deprecation Policy](https://wiki.appcelerator.org/display/guides2/Axway+Appcelerator+Deprecation+Policy) and [Nominal Lifetimes](https://wiki.appcelerator.org/display/guides2/Axway+Appcelerator+Product+Lifecycle#AxwayAppceleratorProductLifecycle-NominalLifetimes) documents for details.
```
[the remark this answers is not in the document]
Comment (sgtcoolguy) [reply changelog/templates/about.hbs]: gotcha

**Comment by sgtcoolguy** [comment]
Yeah, I had installed the cli globally, forgot to add it to the package.json.

Addressed the feedback and pushed back up.

**Review by ewanharris — APPROVED** [review]
Changes look good!

**Pull request merged**